lldb/source/Target/Target.cpp

```diff
@@ -2309,8 +2309,10 @@ Target::GetScratchTypeSystemForLanguage(lldb::LanguageType language,
           m_scratch_type_system_map.RemoveTypeSystemsForLanguage(language);
           type_system_or_err = m_scratch_type_system_map.GetTypeSystemForLanguage(
               language, this, create_on_demand, compiler_options);
-          if (!type_system_or_err)
```

Comment:
This definitely looks correct. Maybe we should use `llvm::scope_exit([&](){GetSwiftScratchContextLock().unlock();});` right at the beginning to cover all paths.

Reply:
done